### PR TITLE
[ISSUE #4483] Standardize exception handling in tests

### LINKED ISSUE #4483: [Enhancement] Standardize exception handling in tests
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Most of the tests in the project were written prior to JUnit 4.13, and use some older idioms instead of using the modern `assertThrows`. Following up on #4475, JUnit Jupiter offers some more elegant options.

### Describe the solution you'd like

- There is no need to catch an exception and explicitly fail the test. The test should be allowed to throw the exception and error-out, which is technically the correct behavior (the test didn't really fail on any logic test).

- In order to test cases when an exception must be thrown, `Assertions.assertThrows` can be used. There is no need to catch the exception assert it is not `null`.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by mureinik**
Fixes #4483.

### Motivation

Following 750825158253d4004e023cfb8ab515c39bc20a71 (#4475), the upgrade to JUnit Jupiter allows us to standardize the way exceptions are handled in tests and reduce boilerplate code.

### Modifications

- There is no need to catch an exception and explicitly fail the test. The test should be allowed to throw the exception and error-out, which is technically the correct behavior (the test didn't really fail on any logic test).
  - In the cases where the only assertion was explicitly failing the test in case an exception was thrown, `Assertions.assertDoesNotThrow` was used instead.
- In order to test cases when an exception must be thrown, `Assertions.assertThrows` can be used. There is no need to catch the exception and assert it's not null.
  - It's also worth noting that some instances that used the idiom of catching an expected exception did not fail the test in case the exception was never thrown, meaning this patch not only improves the tests' style, but also their correctness.
- The original `org.apache.eventmesh.client.grpc.producer.EventMeshGrpcProducerTest#testPublishWithException` test did not in fact test an assertion was thrown, despite the intent hinted by the name. This test was fixed to perform the action it was intended to test.

### Documentation

- Does this pull request introduce a new feature? no
- If yes, how is the feature documented? not applicable
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/admin/controller/ClientManageControllerTest.java [hunk 1]
```diff
@@ -85,13 +85,8 @@ public void testStart() throws Exception {
...
             dummyStatic.when(() -> HttpServer.create(any(), anyInt())).thenReturn(server);
-            try {
-                Mockito.doNothing().when(adminController).run(server);
-                controller.start();
-            } catch (IOException e) {
-                Assertions.fail(e.getMessage());
-            }
-
+            Mockito.doNothing().when(adminController).run(server);
+            controller.start();
```
[the remark this answers is not in the document]
Comment (mureinik) [reply eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/admin/controller/ClientManageControllerTest.java:89]: This was definitely the intention of the original test.

Effectively, even without it the test has the same functionality (if an exception is thrown out of the test method it doesn't pass), but I agree that using `assertDoesNotThrow` would be more explicit. 
I'll amend the PR accordingly.

[the remark this answers is not in the document]
Comment (mureinik) [reply eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/admin/controller/ClientManageControllerTest.java]: @pandaapo  I've pushed an updated version of this PR to address your comment.

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/util/BannerUtilTest.java [hunk 2]
```diff
@@ -17,17 +17,13 @@
...
 
     @Test
     public void testGenerateBanner() {
-        try {
-            BannerUtil.generateBanner();
-        } catch (Exception e) {
-            Assertions.fail("BannerUtil.generateBanner() test failed");
-        }
+        // Must not throw an exception
+        BannerUtil.generateBanner();
```
[the remark this answers is not in the document]
Comment (mureinik) [reply eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/util/BannerUtilTest.java]: fixed

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/util/EventMeshUtilTest.java [hunk 3]
```diff
@@ -167,13 +167,10 @@ public void testCloneObject() throws IOException, ClassNotFoundException {
...
-            ScheduledExecutorService serviceRebalanceScheduler = ThreadPoolFactory
-                .createScheduledExecutor(5, new EventMeshThreadFactory("proxy-rebalance-sch", true));
-            EventMeshUtil.printState((ThreadPoolExecutor) serviceRebalanceScheduler);
-        } catch (Exception e) {
-            Assertions.fail(e.getMessage());
-        }
+        // Must not throw an exception
+        ScheduledExecutorService serviceRebalanceScheduler = ThreadPoolFactory
+            .createScheduledExecutor(5, new EventMeshThreadFactory("proxy-rebalance-sch", true));
+        EventMeshUtil.printState((ThreadPoolExecutor) serviceRebalanceScheduler);
```
[the remark this answers is not in the document]
Comment (mureinik) [reply eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/util/EventMeshUtilTest.java]: fixed

File: eventmesh-sdks/eventmesh-sdk-java/src/test/java/org/apache/eventmesh/client/grpc/producer/EventMeshGrpcProducerTest.java [hunk 4]
```diff
@@ -93,11 +94,7 @@ public void setUp() throws Exception {
 
     @Test
     public void testPublishWithException() {
-        try {
-            producer.publish(defaultEventMeshMessageBuilder().content("mockExceptionContent").build());
-        } catch (Exception e) {
-            assertThat(e).isNotNull();
-        }
+        Assertions.assertDoesNotThrow(() -> producer.publish(defaultEventMeshMessageBuilder().content("mockExceptionContent").build()));
```
[the remark this answers is not in the document]
Comment (mureinik) [reply eventmesh-sdks/eventmesh-sdk-java/src/test/java/org/apache/eventmesh/client/grpc/producer/EventMeshGrpcProducerTest.java]: @pandaapo looking through the history for both this test and the class it's supposed to test, it seems this test never did anything too useful - the call to `publish` just succeeded, and the test implicitly passed without calling any assertion.

I added another commit to explain this and fix the test in the old style by calling `publish` with an argument that's clearly invalid (just a string literal) and asserting that an exception was indeed thrown (see https://github.com/apache/eventmesh/pull/4484/commits/34cd378f5794390d470bd29d5c688dd874684fa5) and then refactored it away to use `assertThrows` in the "main" commit.

**Review by mxsm — APPROVED** [review]
Good Job!! LGTM

**Pull request merged**